src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arr = [...this];
+    let str = arr[0] === null ? '' : arr[0];
+
+    if (!arr.length) {
+      return '';
+    }
+
+    if (arr.length === 1) {
+      return `${str}`;
+    }
+
+    for (let index = 1; index < arr.length; index++) {
+      if (arr[index] === undefined) {
+        str += `${separator}`;
+      } else if (arr[index] === 0 || isNaN(arr[index])) {
+        str += `${separator}${arr[index]}`;
+      } else {
+        str += `${separator}${arr[index] || ''}`;
+      }
+    }
+
+    return str;
```

Comment:
Good work but looks too complicated.
Actually we need only two checks inside the loop.

1. Check if this[i] is not null and is not undefined then you can add this[i] to result string.
2. Check if current index is not last index then you can separator to result string and thats it.